### PR TITLE
Add css variables rules proposal

### LINKED ISSUE #455: Formal definition about how CSS variables work with Airship.
- [x] Review and list all variables
- [x] Define naming rules (BEM?)
- [x] Update developers.md
- [x] Create a interactive acceptance example
- [ ] Update styling guides

### PR DESCRIPTION
**Pull request by IagoLast**
Fix #455 
---

Tasks:

- [x] Add CSS variables DOC
- [x] Add css variables example/smoke
- [x] Review and update all css variables
- [ ] ~~Update css variables docs~~ (I think we should create docs in a different issue, maybe we want to QA this internally first)

Review https://github.com/CartoDB/airship/issues/150

@ivanmalagon  @rjimenezda you can take a look. Tons of changes here but summarizing:

- All variables have a common naming.
- Variables are linked so they always have a valid fallback.
  - Due to this, some fallbacks were removed to reduce codebase-size.
- Some elements have custom variables and other ones can be extended in the future while keeping backward compatibility.

### CODE REVIEW

File: DEVELOPERS.md [hunk 1]
```diff
@@ -162,4 +168,67 @@ Once a new version is released just run `npm run changelog` and push your change
 
 We dont want prereleases to be shown in the changelog so **git tags corresponding to a prerelase should be removed once the QA is done**.
 
-> NOTE: You only need to remove it from github and the `changelog` script will prune all removed tags.
\ No newline at end of file
+> NOTE: You only need to remove it from github and the `changelog` script will prune all removed tags.
+
+## CSS Variables
+
+To achieve a better customization we use [css variables](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_variables). Follow this rules in order to write them in a consistent way.
```
Comment (rjimenezda) [inline DEVELOPERS.md]:
```suggestion
To achieve a better customization we use [css variables](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_variables). Follow these rules in order to write them in a consistent way.
```

File: DEVELOPERS.md [hunk 2]
```diff
@@ -162,4 +168,67 @@ Once a new version is released just run `npm run changelog` and push your change
...
+
+We define two kinds of variables, global and specific:
+
+- **Global Values:** This variables are declared in `styles/core.scss` under the `:host` selector and define specific values that are availiable on every css scope.
```
Comment (rjimenezda) [inline DEVELOPERS.md]:
```suggestion
- **Global Values:** This variables are declared in `styles/core.scss` under the `:root` selector and define specific values that are available on every css scope.
```

File: DEVELOPERS.md [hunk 3]
```diff
@@ -162,4 +168,67 @@ Once a new version is released just run `npm run changelog` and push your change
...
+- **Global Variables:** This variables are declared in `styles/core.scss` under the `:host` selector and are availiable on every css scope.
```
Comment (rjimenezda) [inline DEVELOPERS.md]:
```suggestion
- **Global Variables:** This variables are declared in `styles/core.scss` under the `:root` selector and are available on every css scope.
```

File: DEVELOPERS.md [hunk 4]
```diff
@@ -162,4 +168,67 @@ Once a new version is released just run `npm run changelog` and push your change
...
+- **Specific Values:** This variables are declared in the element/component styles and define a value that can be used in different places, usually specific variables. (Same value could be used in the table border and paragraph color)
+- **Specific Variables:** This variables are defined in the stylss of element/component and they only are availiable under the element/component scope.
```
Comment (rjimenezda) [inline DEVELOPERS.md]:
```suggestion
- **Specific Variables:** This variables are defined in the styles of element/component and they're only  available under the element/component scope.
```

File: DEVELOPERS.md [hunk 5]
```diff
@@ -162,4 +168,67 @@ Once a new version is released just run `npm run changelog` and push your change
...
+
+The variables should have a less specific one as fallback, so for example when developing the `histogram` widget:
```
Comment (rjimenezda) [inline DEVELOPERS.md]:
```suggestion
The variables should have a less specific one as fallback, so for example when developing the `stacked-bar` widget:
```

File: DEVELOPERS.md [hunk 6]
```diff
@@ -162,4 +168,67 @@ Once a new version is released just run `npm run changelog` and push your change
...
+```css
+
+  /** <style/core/core.scss> The values are initialized **/
+  --as--color--primary: white;
+
+  /** <stacked-bar-widget.scss> The values used with chained fallbacks, last fallback is a css-value **/
+  --as-stacked-bar-widget--background-color: var(--as--widget__header--background-color, --as--color--primary);
+```
+
+Doing this users can controll specific widgets `--stacked-bar-widget--background-color` share a style for all widgets `--as--widget__header--background-color` or just redefine the default value `--as--color--primary`.
```
Comment (rjimenezda) [inline DEVELOPERS.md]: This paragraph is confusing, I rewrote it but I don't know if it's exactly what you meant.

```suggestion
With `--stacked-bar-widget--background-color` users can control specific widgets, with the shared variable `--as--widget__header--background-color` they can control all widgets, or they can just redefine the default value `--as--color--primary`.
```

By the way, the class is `--widget-background-color` right now, not `--as--widget__header--background-color`. I don't know if you plan to change it, just noticed it.

Comment (IagoLast) [reply DEVELOPERS.md]: my idea is to change them to follow the rules :)

Comment (rjimenezda) [reply DEVELOPERS.md]: great

**Pull request merged**